src/main/java/io/vertx/serviceproxy/generator/ServiceProxyGen.java

```diff
@@ -167,25 +183,47 @@ else if (t.getKind() == ClassKind.DATA_OBJECT)
       writer.stmt("_json.put(\"" + name + "\", " + name + ")");
   }
 
+  private void generateSendCallWithFutureReturn(TypeInfo returnType, CodeWriter writer) {
+    ParameterizedTypeInfo parameterizedTypeInfo = ((ParameterizedTypeInfo)returnType);
+    TypeInfo t = parameterizedTypeInfo.getArg(0);
+    String typeParams = parameterizedTypeInfo.getArgs().stream().map(TypeInfo::getSimpleName).collect(Collectors.joining(","));
+    writer.format("Promise<%s> promise = Promise.promise();", typeParams).println();
```

Comment:
@vietj The generated `Promise` here should be created starting from the context?

Reply:
@slinkydeveloper There are different implementations in 4.0，[see it](https://github.com/vert-x3/vertx-service-proxy/pull/109/files#diff-35042b716e99f2f7e4c85383ada7fc00R227-R237)